### PR TITLE
Sync NOTICE file with repository licenses

### PR DESCRIPTION
**Pull request by danielgustafsson**
This moves the HyperLogLog license from THIRDPARTY into the correct NOTICE file, and removes the world database license as we no longer vendor that (it was part of BugBuster). Also fixes a few related typos.

### CODE REVIEW

**Comment by hlinnaka** [comment]
> ..., and removes the world database license as we no longer vendor that (it was part of BugBuster)

There are still remnants of it in other tests. Try: `git grep lifeexpectancy`.

**Comment by danielgustafsson** [comment]
> > ..., and removes the world database license as we no longer vendor that (it was part of BugBuster)
> 
> There are still remnants of it in other tests. Try: `git grep lifeexpectancy`.

Hrmpf, I was apparently not casting my grep-net wide enough, missed that. PR updated with the license re-added, thanks!

File: NOTICE [hunk 1]
```diff
@@ -1942,7 +1942,7 @@ Network Circle, Santa  Clara, California 95054, U.S.A.
 *  4. Use of this program or creation of derived works based on this
 *     program constitutes acceptance of these licensing restrictions.
 *  5. Absolutely no warranty is expressed or implied.
-*----------------------------------------------------------------
+*-----------------------------------------------------------------------
```
Comment (hlinnaka) [inline NOTICE]: These lines are part of the upstream license blurp, I think. Also,  if you shorten them, then the text in the license will go beyond the lines. Would be better to leave it alone, I think.

Comment (danielgustafsson) [reply NOTICE]: They are, and that's why they're included as we had shortened them. Including them makes the license copy verbatim.

Comment (hlinnaka) [reply NOTICE]: Oh, I read the diff backwards! Yes, +1 on expanding them to match upstream again.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks for review

**Pull request closed without merging**